sapp/ui/server.py

```diff
@@ -99,5 +99,12 @@ def start_server(
     if static_resources:
         application.static_folder = static_resources
     if debug:
-        CORS(application, resources={r"/graphql": {"origins": "http://localhost:3000"}})
+        default_port = int(os.environ.get("PORT"), 10) or 3000
```

Comment:
> My only concern with this current implementation is that the behaviour of parseInt() in JavaScript may differ from Python's int() function for specific edge cases and may result in unexpected issues.

Like you mentioned, `int(os.environ.get("PORT"), 10)` here will result in a `TypeError`. The suggestion below should fix that:
```suggestion
        default_port = os.environ.get("PORT") or 3000
```